docs/docs/state.md

```diff
@@ -45,37 +43,39 @@ class Demo extends Component {
 
 ## State flows
 
-The most typical flow of the state is when the gesture handler first picks up the initial touch events, then at some point it recognizes the touches and after the end of the gesture is recognized it resets back to the initial state. The flow looks as follows (longer arrows represent that there are possibly more touch events received before the state changes):
+The most typical flow of state is when a gesture handler picks up on an initial touch event then recognizes it then acknowledges its ending then resets itself back to the initial state. 
+
+The flow looks as follows (longer arrows represent that there are possibly more touch events received before the state changes):
 
 [`UNDETERMINED`](#undetermined) -> [`BEGAN`](#began) ------> [`ACTIVE`](#active) ------> [`END`](#end) -> [`UNDETERMINED`](#undetermined)
 
-Here is another possible flow when the handler receive touches that causes it to fail recognition:
+Another possible flow is when a handler receives touches that cause a recognition failure:
 
 [`UNDETERMINED`](#undetermined) -> [`BEGAN`](#began) ------> [`FAILED`](#failed) -> [`UNDETERMINED`](#undetermined)
 
-Lastly even when the handler properly recognizes the gesture it may be interrupted by the touch system, in that case the flow looks as follows:
+Another possible flow is when a handler *does* properly recognize the gesture but then is interrupted by the touch system causing it to cancel. In that case the flow looks as follows:
 
 [`UNDETERMINED`](#undetermined) -> [`BEGAN`](#began) ------> [`ACTIVE`](#active) ------> [`CANCELLED`](#cancelled) -> [`UNDETERMINED`](#undetermined)
 
 ## States
 
-Section below lists all the possible handler states along with detailed description of each state:
+The section below describes all possible handler states:
 
 ### UNDETERMINED
-This is the initial state of each handler. It also changes to that state when it is done recognizing and hasn't started recogninzing another gesture.
+This is the initial state of each handler. It can be found before gestures have occurred and after they've completed. 
 
 ### FAILED
-Handler has received some touches, but for some condition (e.g. finger traveled too long distance when `maxDist` property is set) it won't get [activated](#ACTIVE) and gesture was not recognized. After that handler is reset to [the initial state](#undetermined).
+A handler has received some touches but for some reason won't recognize them. Its state will **not** become `ACTIVE` but instead become `FAILED`. Afterwards, it is reset to `UNDETERMINED`. An example of this is if a finger travels more distance than a defined `maxDist` property allows.
```

Comment:
```suggestion
A handler has received some touches but for some reason didn't recognize them. An example of this is if a finger travels more distance than a defined `maxDist` property allows. In that situation, the handler won't become active and will fail instead. Afterward, it's state will be reset from to `UNDETERMINED`. 
```

"Afterward, it is reset to `UNDETERMINED`" – I think this kinda drifts away from a state machine abstraction.

I'm not sure about this suggestion, it surely can be worded better.

Reply:
I will make a small change to this, please review